### ISSUE TITLE
[Bug] CAE problem raised by Azure SDK

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#4165]
MSAL 4.54.1

Scenario: 

1. A distributed system with shared cache uses MSAL. OR several apps, sharing the client_id and cache (this is more Az SDK case - although they don't have different apps)
2. Some of the apps are CAE capable and some are not. 
3. An app that is CAE capable gets a token for Graph. The token will have a special claim `cp1`
4. A client that is not CAE capable needs a token for Graph. It gets the token with `cp1` claim from the cache. 
5. The non-CAE capable client makes the call to Graph. 
6. Graph decides to not give access and issues a claims challenge back to the client

**Actual**: the client crashes or loops, because it can't handle the claims challenge

Possible solutions to this: 

- MSAL can scope access tokens by claims (cp1 is a claim). This would separate the access tokens in the cache into cp1 tokens and non-cp1 tokens. A non-CAE capable client trying to acquire a token from the cache will not be able to retrieve cp1 tokens.
Some MSALs already use this but there are downsides.

CC @jmprieur @localden @rayluo @gladjohn

**Comment by rayluo** [comment]
> MSAL 4.54.1
> 
> Scenario:
> 
>     1. A distributed system with shared cache uses MSAL. OR several apps, sharing the client_id and cache (this is more Az SDK case - although they don't have different apps)
> 
>     2. Some of the apps are CAE capable and some are not.
> 
>     3. An app that is CAE capable gets a token for Graph. The token will have a special claim `cp1`
> 
>     4. A client that is not CAE capable needs a token for Graph. It gets the token with `cp1` claim from the cache.
> 
>     5. The non-CAE capable client makes the call to Graph.
> 
>     6. Graph decides to not give access and issues a claims challenge back to the client

So, that scenario could happen on all MSALs, not just MSAL .Net 4.54.1.

> > **Actual**: the client crashes or loops, because it can't handle the claims challenge
>
> we should not handle this at MSAL level. I think it's a deployment issue.

Agreed with this assessment. This does not seem a bug at MSAL level.

I do not quite get the "version 1.0, 1.1, 1.2" setup, but perhaps we can just suggest that "even if your CAE-capable app and CAE-incapable app may share the same client_id, they shall not share the same token cache"?

Ideally, our CAE documentation shall be written in a language-agnostic way, so that all ID4S MSALs can reference to the same doc.

By the way, the other day Azure SDK team comtemplated having CAE-capability specified in a per-request basis. Was it from the same scenario? Per-request CAE capability does not seem to help in this scenario, though.

**Comment by bgavrilMS** [comment]
Yes, same scenario, but it won't solve their issue. They need 2 instances of PCA / CCA, so that they can point at different caches.
